heareval/tasks/util/luigi.py

```diff
@@ -100,32 +104,33 @@ def stage_number(self) -> int:
         return 0
 
 
-class ExtractCorpus(WorkTask):
-
+class ExtractZipFile(WorkTask):
+    
     infile = luigi.Parameter()
-
+    
     @property
     def name(self):
         return type(self).__name__
 
-    def run(self):
+    def requires(self):
+        #Some Task should be returned having a working directory from 
+        #which the infile will be picked.
+        raise NotImplementedError
 
-        filename, extension = os.path.split(self.infile)
+    def run(self):
+        #Assert if the previous task has a workdir from which the file will be picked.
+        assert hasattr(self.requires(), "workdir")
 
         # TODO -- run correct extraction given archive type
-        print(extension)
-
         # Location of zip file to extract. Figure this out before changing
         # the working directory.
         corpus_zip = os.path.realpath(
-            os.path.join(self.requires().workdir, "corpus.zip")
+            os.path.join(self.requires().workdir, self.infile)
         )
-        subprocess.check_output(["unzip", "-o", corpus_zip, "-d", self.workdir])
-
+        shutil.unpack_archive(corpus_zip, self.workdir) 
```

Comment:
Using `shutil.unpack_archive`. I checked it can automatically infer the type of the zip file and unzip.

Reply:
Nice